_pages/dev/ssr/server-side-rendering-error-handling.md

```diff
@@ -0,0 +1,349 @@
+    : join(distFolder, 'index.html');
+  const indexHtmlContent = readFileSync(indexHtml, 'utf-8');
+
+  /* ... */
+
+  server.use(customErrorPageErrorHandlers); // <-- custom error handler for 404 provided before default error handlers.
+  server.use(defaultExpressErrorHandlers(indexHtmlContent));
+
+  return server;
+}
+```
+
+## Strict error handling in Angular for HTTP and NgRx
+
+ To seal the Angular application from the asynchronous errors that occur in NgRx flow and HTTP calls to backend during the rendering process, Spartacus provides set of tools to handle them.
+
+### Handling HTTP errors
+
+To handle HTTP errors, Spartacus provides `HttpErrorHandlerInterceptor` that catches the HTTP errors and forward them to the `CxErrorHandler`. The interceptor by default helps to distinguish two types of error by wrapping them with custom error classes before passing them to the `CxErrorHandler`:
+- `OutboundHttpError` - represents an outbound HTTP error that occurs when communicating with the backend.
+- `CmsPageNotFoundOutboundHttpError` - represents an outbound HTTP error specific to a CMS page not found. Extends the base OutboundHttpError class.
+
+Thanks to this, when such semantic errors are propagated to ExpressJS and handled there by `defaultExpressErrorHandlers` middleware, it can distinguish the errors and return an appropriate status code to the client (e.g. 404 or 500).
+
+**Note:** Make sure your custom Angular HTTP interceptors are provided after `HttpErrorHandlerInterceptor` (so they run as next in the sequence). If they catch any error, make sure to rethrow the error downstream to next interceptors. Otherwise, Spartacus `HttpErrorHandlerInterceptor` will not capture such an error, and therefore it won't be able to propagate it to ExpressJS layer and cause sending and appropriate HTTP error status code to the client of the SSR.  
+
+Spartacus `HttpErrorHandlerInterceptor` treats all outbound HTTP as a reason to make SSR fail. 
+
+However, if this assumption is not true for your application, you can customize it by over-providing Spartacus `HttpErrorHandlerInterceptor`. It might be useful in one of the following cases:
+- when you don't want some backend HTTP errors to make SSR fail (e.g. when some HTTP errors from backend are considered really not important for the final HTML result of SSR)
+- when you want some backend HTTP success responses to be treated as errors making SSR to fail (e.g. when backend responds with status 200 and a custom payload indicating and error like `{ "ok": false, ... }`
+
+### Handling NgRx errors
+
+Spartacus treats any NgRx action containing the property `error` as a reason for SSR to fail. To handle them, Spartacus provides elements of NgRx flow that helps to catch the errors and forward them to the `CxErrorHandler`:
+- `CxErrorHandlerEffect` - an effect that catches all dispatched NgRx Actions with the property `error` and forwards such an error to the `CxErrorHandler`.
```

Comment:
`CxErrorHandlerEffect` is deliberately a part of our Private API.
What customers can see and customize is `ErrorActionService`. So what about:

```suggestion
- `ErrorActionService` - it filters all dispatched NgRx Actions with the property `error` and forwards such an error to the `CxErrorHandler`.
```